bsp/src/org/jetbrains/bsp/project/BspTestRunLineMarkerProvider.scala

```diff
@@ -0,0 +1,34 @@
+package org.jetbrains.bsp.project
+
+import com.intellij.execution.lineMarker.RunLineMarkerContributor
+import com.intellij.openapi.module.ModuleManager
+import com.intellij.psi.{PsiClass, PsiElement}
+import org.jetbrains.bsp.data.BspMetadata
+import org.jetbrains.plugins.scala.testingSupport.test.ScalaTestRunLineMarkerProvider
+
+import scala.collection.JavaConverters._
+
+class BspTestRunLineMarkerProvider extends ScalaTestRunLineMarkerProvider {
```

Comment:
- This class is mostly a dummy one. Probably it does not have to extends `ScalaTestRunLineMarkerProvider`. 
- Gutter icons are just dummy ones. 
- `isBspTestClass` is very inefficient. It iterates over all modules, gets all `BspMetadata` objects and iterates over all test classes associated with them. 
- `name.contains(tc) || tc.contains(name)` is broken. I need a better way to handle this. Qualified class name can include $ chars (when it's a scala object),  but BSP provides clean test class names (maybe that's a bug in mill 😕)

Reply:
If it's inefficient we'll have to estimate performance impact on projects with many tests, both performance-wise and memory-load-wise. There might be some point where it's more effective to request classes on demand?

Reply:
How do the test class names look respectively, what's the reverse `contains` need?

Reply:
How does this provider interact when we have local support for tests as well?

Reply:
I am not sure what you mean by local support for tests. Before these changes there were no gutter icons for tests. After this commit, there are gutter icons clicking which triggers some UTest runner. So I don't get why the gutter icons were not previously displayed if intellij had UTest support. Maybe I need to configure it somehowe ?

Reply:
I see that `ScalaTestRunLineMarkerProvider` is actually returning null for some of the supported test frameworks. Maybe Roman Shein can tell in more detail why.